rfcs/0000-driver-registry.md

```diff
@@ -0,0 +1,379 @@
+# RFC - Drivers and Discovery
+
+## Goal
+
+MnemOS should have some way of "automatically discovering" drivers. This would be used by three main "consumers":
+
+* The kernel itself, which may need to use drivers that are provided at initialization time
+* Drivers, which may depend on other drivers (such as a virtual serial mux driver, which depends on a "physical serial port" driver)
+* Userspace, which will typically interact with drivers through either the stdlib, or other driver crates.
+
+## Background
+
+Right now, when spawning drivers, the process goes roughly as follows:
+
+* The `init()` code manually constructs drivers, and manually creates any shared resources necessary between drivers
+* The `init()` code spawns the driver task(s) onto the kernel executor
+
+This is a little manual, and also won't really work for userspace.
+
+There are also some half baked ideas in the `abi`/`mstd` crates about having a single "message kind" that is an enum of all possible driver types, but this isn't very extensible
+
+## The proposal
+
+This change proposes inntroducing a **registry** of drivers, maintained by the kernel.
+
+* This **registry** would be based on a Key:Value type arrangement:
+    * The **Key** would be a tuple of:
+        * A pre-defined **UUID**, that uniquely defines a "service" that the driver fulfills, e.g. the ability to process a given command type
+        * The `TypeId` of that message
```

Comment:
The intent behind this is to allow a single driver (with a single UUID) to handle multiple message types?